EpistasisLab/tpot2
# Random state fixes

[please review the [Contribution Guidelines](http://epistasislab.github.io/tpot/contributing/) prior to submitting your pull request. go ahead and delete this line if you've already reviewed said guidelines.]

## What does this PR do?

I changed the `random_state=#` variable for the cv/data splits to a user-specified random seed. This gives the user more control over reproducibility, as far as the splitting of data goes.

## Where should the reviewer start?

I have added a `seed`  argument to be passed into the constructor for both estimator classes. This is a good place to start. In both of these classes `random_state=#` has been set to `random_state=self.seed`. Additionally, the `self.seed` variable is also passed to the `objective_function_generator` due to it splitting data and using `random_state`. The `estimator_utils.py` files show how we pass the seed to the `objective_function_generator`.

## How should this PR be tested?

We can just run normal tests to make sure nothing breaks. This is only to increase user reproducibility for experiments. More will be added in the future in regard to this, but this is a good first step in that direction.

## Any background context you want to provide?

Previously, the `random_state` variable was set to a hardcoded number. This results in the data being split the same every time an experiment is done. I think by allowing users to specify a seed, we can increase the generality of our results while controlling the randomness better. 
 
## What are the relevant issues?

[you can link directly to issues by entering # then the number of the issue]

## Screenshots (if appropriate)



## Questions:

- Do the docs need to be updated?
 
Maybe? We can mention that a seed can be passed but will default to None if not.

- Does this PR add new (Python) dependencies?

No.
